spec/services/school_data_importer_spec.rb

```diff
@@ -0,0 +1,132 @@
+require "rails_helper"
+require "csv"
+
+RSpec.describe SchoolDataImporter do
+  let(:school_data_importer) { SchoolDataImporter.new }
+
+  describe "#run" do
+    let(:date_string) { Time.zone.now.strftime("%Y%m%d") }
+    let(:file) { spy(:file) }
+    let(:gias_csv_url) { "http://ea-edubase-api-prod.azurewebsites.net/edubase/edubasealldata#{date_string}.csv" }
+    let(:csv_file_name) { "tmp/schools_data_#{date_string}.csv" }
+    let(:row) do
+      {
+        "URN" => "12345",
+        "EstablishmentName" => "Penistone Grammer School",
+        "Street" => "123 High Street",
+        "PhaseOfEducation (code)" => "4",
+        "LA (code)" => "40",
+        "LA (name)" => "Blackpool",
+        "EstablishmentTypeGroup (code)" => "10",
+        "TypeOfEstablishment (code)" => "22",
+      }
+    end
+    let!(:request) { stub_request(:get, gias_csv_url).to_return(body: "foo", status: 200) }
```

Comment:
We can make this spec more of an integation spec and less brittle to changes of the internal implementation if we return a subset of actual CSV data here from the request. That would remove the need to stub `File` and `CSV` and make it so all the refactoring suggested above could be performed without having to change this spec.